### DISCUSSION TITLE
Suitable module to run AtomLight

### DISCUSSION THREAD

**Discussion by wingnut5959** [PlummersSoftwareLLC/NightDriverStrip#597] [Q&A]
I've had a tough time finding a module that is capable of running the AtomLight project. Has anyone been successful in getting this going and able to share which module they used to run it on? This [one](https://www.amazon.com/Microcontroller-ESP32-S3-WROOM-1-N8R8-Development-Processor-MicroPython/dp/B0C28FFBRP/ref=sr_1_3?crid=17I95Q4EOM437&keywords=esp32+8mb+psram&qid=1706032263&s=electronics&sprefix=esp32+8m%2Celectronics%2C377&sr=1-3) appears to meet the requirements but not completely sure.

**Comment by prschguy1** [comment]
I have had that one running pretty well on M5stack. It runs pretty well on 4 channel, although you'll have to expose some extra pins by removing the extension board via jumper leads. So far, that chip will carry just a little short of 77 effects. Haven't tried it on the current code base, but it used to work anyway. Esp32-s3 builds aren't currently working all that well to be honest. I am having a struggle managing just a few effects. Still things to be worked out there. I'm using the 32 mb chip, and still running out of memory when running a large amount of effects. Wifi is pretty stable on M5stack until you really load it up.  Just my 2 cents though. If you elect to use that one, will need the groove connector to gain access to all the pins.

Craig

**Reply by wingnut5959** [reply]
Thanks for your input. There are so may M5 products that fall under M5stack, can you be more specific? I was able to build and upload to a d1 mini 32, the system just goes into a crash/reboot cycle. I'm guessing not enough memory on that board.

**Comment by prschguy1** [comment]
Will agree with the you there. The m5stack has 16mb of memory as well as psram. So far, has been a true performer here.  The technology has been moving so quickly here. The last board I have used here is the M5stack core 2.  It has served me well.  Have tried d1 mini, and just doesn't  seem to have enough memory.

**Comment by prschguy1** [comment]
Boot looping, now this is disturbing. I have not built atomlight in a while. It used to work correctly on d1 mini. I just tried it out, and agree, it is boot looping now. I also just uploaded it onto heltec v1, the board it was originally built for, and that also boot loops. You should probably open up an issue on this so someone can look at it. I reduced d1 to 2 channel, and it runs pretty stable, but should be able to handle 4 channel. I wonder now how m5stack is doing. This is the m5stack I used. Keep in mind: not all effects are sound reactive. Only some of the effects are multichannel. only some multi channel effects look right on 4 channel. Some effects work better on 2 channel on 
 2 opposite spokes of atomlight


![20240124_040121_resized](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/69419979/70dce993-0295-4805-9483-78dad74c7b6a)



![20240124_040215_resized](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/69419979/ad7e9a89-c4a6-49cc-a789-4ea0abea8b59)


Craig
